bin/alluxio-process

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env bash
+#
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+set -e
+
+BIN=$(cd "$( dirname "$( readlink "$0" || echo "$0" )" )"; pwd)
+
+USAGE="Usage: alluxio-process <process name> [-c]
+
+alluxio-process launches one of the Alluxio processes in the foreground.
+By default it will only log to the configured log file. To configure the
+process, add configuration properties in alluxio-site.properties or
+environment variables in conf/alluxio-env.sh.
+
+Note that this script does not verify Alluxio configuration before
+starting. It simply populates the necessary environment variables and
+launches the desired process in the foreground. It is recommended to use
+alluxio-start.sh unless you know what you are doing.
+
+    <process name>    One of: master, secondary_master, job_master, worker,
+                      job_worker, proxy, logserver
+
+    -c                Provide this flag to also log directly to stdout
+
+    -h                Print this help text to stdout
+"
+
+# Launches an Alluxio process
+#
+# Arguments:
+#         1: name of the *_JAVA_OPTS variable to use when launching
```

Comment:
it's uncommon to take the var name as the argument and use it using "reflection"?,
can we quote the list of opts to pass as one single arg?

Reply:
Variable indirection is also a well documented feature of bash variables. I would not call it "reflection". Is there a technical limitation, or something else you don't like about it?

Either way, without having the variable value of the variable passed as an arg, then the following conditional check would not be possible without duplication:

https://github.com/Alluxio/alluxio/blob/3c3f36b2360d47aed24c5e308e376b64a6fc873b/bin/alluxio-process#L51-L53

Reply:
After offline discussion, we decide that the behavior for setting all `ALLUXIO_*_JAVA_OPTS` variables is done within `./libexec/alluxio-config.sh` and it should be this script's responsibility to do so.

With that assumption, we can remove all of these checks before starting the processes. It has been documented within the script.